source/postprocess/visualization/vertical_heat_flux.cc

```diff
@@ -0,0 +1,117 @@
+
+        MaterialModel::MaterialModelInputs<dim> in(n_quadrature_points,
+                                                   this->n_compositional_fields());
+        MaterialModel::MaterialModelOutputs<dim> out(n_quadrature_points,
+                                                     this->n_compositional_fields());
+
+        //Create vector for the temperature gradients.  All the other things
+        //we need are in MaterialModelInputs/Outputs
+        std::vector<Tensor<1,dim> > temperature_gradient(n_quadrature_points);
+
+        in.position = evaluation_points;
+        in.strain_rate.resize(0); // we do not need the viscosity
+        for (unsigned int q=0; q<n_quadrature_points; ++q)
+          {
+            in.pressure[q]=uh[q][this->introspection().component_indices.pressure];
+            in.temperature[q]=uh[q][this->introspection().component_indices.temperature];
+            for (unsigned int d = 0; d < dim; ++d)
+              {
+                in.velocity[q][d]=uh[q][this->introspection().component_indices.velocities[d]];
+                in.pressure_gradient[q][d] = duh[q][this->introspection().component_indices.pressure][d];
+                temperature_gradient[q][d] = duh[q][this->introspection().component_indices.temperature][d];
+              }
+
+            for (unsigned int c=0; c<this->n_compositional_fields(); ++c)
+              in.composition[q][c] = uh[q][this->introspection().component_indices.compositional_fields[c]];
+          }
+
+        this->get_material_model().evaluate(in, out);
+
+        for (unsigned int q=0; q<n_quadrature_points; ++q)
+          {
+            const Tensor<1,dim> gravity = this->get_gravity_model().gravity_vector(in.position[q]);
+            const Tensor<1,dim> vertical = -gravity/( gravity.norm() != 0.0 ?
+                                                      gravity.norm() : 1.0 );
+            const double advective_flux = (in.velocity[q] * vertical) * in.temperature[q] *
+                                          out.densities[q]*out.specific_heat[q];
+            const double conductive_flux = -(temperature_gradient[q]*vertical) *
+                                           out.thermal_conductivities[q];
+            computed_quantities[q](0) = advective_flux + conductive_flux;
+          }
+      }
+    }
+  }
+}
+
+
+// explicit instantiations
+namespace aspect
+{
+  namespace Postprocess
+  {
+    namespace VisualizationPostprocessors
+    {
+      ASPECT_REGISTER_VISUALIZATION_POSTPROCESSOR(VerticalHeatFlux,
+                                                  "vertical heat flux",
+                                                  "A visualization output object that generates output "
+                                                  "for the heat flux in the vertical direction, with "
+                                                  "the sign convention of positive flux upwards.")
```

Comment:
Can you clarify this by saying that it is the *total* heat flux, i.e., conductive plus convective?